### ISSUE TITLE
Need a mechanism to enable DAWR on future CPUs

### ISSUE THREAD

**Issue by mpe** [linuxppc/linux#180]
As documented in c6bef2e9e50ca27987aae90147511bd320a2264e, the DAWR (Data Address Watchpoint Register) is disabled on Power9 due to an unpublished errata.

That is OK except that we have no mechanism for firmware to tell us to enable it on a future CPU where the errata is fixed. That means the only way to enable it on such a CPU is to update the kernel.

**Comment by leitao** [comment]
Hi Michael,

I am trying to understand what are the options that we could use to solve this problem. Right now I see only one possible way:

Definining a firmware property that will let us know if DWAR is broekn or not for that level, and assume that is is broken if the property is not set.

Is this what you thought about?

**Comment by mpe** [comment]
Yeah with skiboot we have the `ibm,powerpc-cpu-features` node(s) which is how we should solve it there.

On PowerVM / KVM we need to do something different. PAPR defines a `ibm,pa-features` property, which is probably the right way to do it. But we'd need to talk to the PAPR people.
